examples/nfts.ts

```diff
@@ -0,0 +1,175 @@
+import { $, alice, bob, Event as CapiEvent, Rune, ValueRune } from "capi"
+import { signature } from "capi/patterns/signature/nft.ts"
+import { Nfts, WestmintLocal } from "zombienet/nfts.toml/collator/@latest/mod.js"
+import { Event } from "zombienet/nfts.toml/collator/@latest/types/pallet_nfts/pallet.js"
+import { MintType } from "zombienet/nfts.toml/collator/@latest/types/pallet_nfts/types.js"
+import { RuntimeEvent } from "zombienet/nfts.toml/collator/@latest/types/westmint_runtime.js"
+
+// The nfts pallet uses inverted bitflags; on means exclude and off means include.
+const DefaultCollectionSetting = {
+  TransferableItems: 1n << 0n,
+  UnlockedMetadata: 1n << 1n,
+  UnlockedAttributes: 1n << 2n,
+  UnlockedMaxSupply: 1n << 3n,
+  DepositRequired: 1n << 4n,
+  allOff: 0n,
+  allOn: 0b11111n,
+}
+const DefaultItemSetting = {
+  Transferable: 1n << 0n,
+  UnlockedMetadata: 1n << 1n,
+  UnlockedAttributes: 1n << 2n,
+  allOff: 0n,
+  allOn: 0b111n,
+}
+
+const createCollection = await Nfts
+  .create({
+    config: Rune.rec({
+      settings: DefaultCollectionSetting.allOn,
+      mintSettings: Rune.rec({
+        mintType: MintType.Issuer(),
+        defaultItemSettings: DefaultItemSetting.allOn,
+      }),
+    }),
+    admin: alice.address,
+  })
+  .signed(signature({ sender: alice }))
+  .sent()
+  .dbgStatus("Create Collection:")
+  .finalizedEvents()
+
+const getCollectionIdFromEvents = (events: CapiEvent<WestmintLocal>[]) => {
+  const event = events.find((event) =>
+    RuntimeEvent.isNfts(event.event) && Event.isCreated(event.event.value)
+  )?.event.value as Event.Created | undefined
+  return event?.collection
+}
+
+// Create a collection and extract its id from the events
+const collection = await createCollection
+  .into(ValueRune)
+  .map(getCollectionIdFromEvents)
+  .unhandle(undefined)
+  .dbg("Collection Id:")
+  .run()
+
+// Create and encode metadata for the collection
+const collectionMetadata = $.object($.field("name", $.str)).encode({ name: "Collection 01" })
+
+// Set the collection's metadata
+await Nfts
+  .setCollectionMetadata({
+    collection,
+    data: collectionMetadata,
+  })
+  .signed(signature({ sender: alice }))
+  .sent()
+  .dbgStatus("Set Collection Metadata:")
+  .finalized()
+
+const item = 0
+
+// Mint an item to the collection
+Nfts
+  .mint({
+    collection,
+    item,
+    mintTo: alice.address,
+  })
+  .signed(signature({ sender: alice }))
+  .sent()
+  .dbgStatus("Mint Item:")
+  .finalized()
+
+// Create and encode metadata for the NFT
+const itemMetadata = $.object($.field("name", $.str)).encode({ name: "NFT #01" })
+
+// Set the NFT's metadata
+await Nfts
```

Comment:
you can probably batch some of these calls. i.e set metadata and price can go together